src/main.jsx

```diff
@@ -0,0 +1,514 @@
+      this.populateInitialAppState();
+    }
+  }
+
+  render() {
+    return (
+      <div className={this.state.dataInitialized ? 'container-full' : 'container-full loading'}>
+
+        <Navigation
+          doStatsLinkClick={this.handleStatsLinkClick}
+          doFindProbesLinkClick={this.handleResetToDefaultView}
+          datePublished={this.props.generalFetch.value}
+        />
+
+        <SearchForm
+          {...this.props}
+          versions={this.state.versions}
+          channels={CHANNELS}
+          showReleaseOnly={this.state.showReleaseOnly}
+
+          doChannelChange={this.handleChannelChange}
+          doShowReleaseOnlyChange={this.handleShowReleaseOnlyChange}
+          doProbeConstraintChange={this.handleProbeConstraintChange}
+          doVersionChange={this.handleVersionChange}
+          doSearchConstraintChange={this.handleSearchConstraintChange}
+          doSearchTextChange={this.handleSearchTextChange}
+          activeView={this.state.activeView}
+
+          searchText={this.state.searchText}
+          selectedChannel={this.state.selectedChannel}
+          selectedSearchConstraint={this.state.selectedSearchConstraint}
+          selectedProbeConstraint={this.state.selectedProbeConstraint}
+          selectedVersion={this.state.selectedVersion}
+        />
+
+        <ProbeDetails
+          selectedProbe={this.state.selectedProbe}
+          channelInfo={this.state.channelInfo}
+          revisions={this.props.revisionsFetch.value}
+          selectedChannel={this.state.selectedChannel}
+          datasets={this.props.datasetsFetch.value}
+          doCloseProbeDetails={this.handleResetToDefaultView}
+          activeView={this.state.activeView}
+        />
+
+        <Stats
+          selectedProbeConstraint={this.state.selectedProbeConstraint}
+          selectedChannel={this.state.selectedChannel}
+          channelInfo={this.state.channelInfo}
+          probes={this.state.allProbes}
+          revisions={this.props.revisionsFetch.value}
+          dataInitialized={this.state.dataInitialized}
+          activeView={this.state.activeView}
+        />
+
+        <SearchResults
+          channelInfo={this.state.channelInfo}
+          probes={this.state.probes}
+          revisions={this.props.revisionsFetch.value}
+          selectedChannel={this.state.selectedChannel}
+          dataInitialized={this.state.dataInitialized}
+          doExposeProbeDetails={this.handleExposeProbeDetails}
+          activeView={this.state.activeView}
+        />
+
+        <Spinner />
+
+      </div>
+    );
+  }
+}
+
+export default connect(() => ({
+  generalFetch: process.env.REACT_APP_DATA_GENERAL,
+  revisionsFetch: process.env.REACT_APP_DATA_REVISIONS,
+  probesFetch: process.env.REACT_APP_DATA_PROBES,
+  environmentFetch: process.env.REACT_APP_DATA_ENVIRONMENT,
+  otherFieldsFetch: process.env.REACT_APP_DATA_OTHER_FIELDS,
+  datasetsFetch: process.env.REACT_APP_DATA_DATASETS
+}))(Main);
```

Comment:
Do you think it would be over-optimizing to split this file up? It's pretty big, but I can be convinced otherwise.